torchft/manager.py

```diff
@@ -470,29 +472,52 @@ def _async_quorum(
             self._pg.configure(store_prefixed_addr, replica_rank, replica_world_size)
             self._quorum_id = quorum_id
 
-        # See manager.rs for healing conditions
-        if heal and allow_heal:
-            self._healing = True
-            self._logger.info(
-                f"healing required, fetching checkpoint server address from {address=} {max_step=}"
-            )
-            primary_client = ManagerClient(
-                address, connect_timeout=self._connect_timeout
-            )
-            checkpoint_server_address = primary_client.checkpoint_address(
-                self._rank, timeout=self._timeout
-            )
+        if allow_heal:
+            if quorum.recovering_ranks:
```

Comment:
the `if recovering_ranks` condition is always true if `if heal` is true as well right?

Reply:
They're flipped, this checks if anyone will be requesting a checkpoint from this node where as `heal` below is whether this node will be requesting a checkpoint

`heal == quroum.recover_rank is not None` which is distinct from `quorum.recovering_ranks`